src/Symfony/Component/Yaml/Tests/ParserTest.php

```diff
@@ -803,6 +803,84 @@ public function testMappingDuplicateKeyFlow()
         $this->assertSame($expected, Yaml::parse($input));
     }
 
+    /**
+     * @dataProvider getParseExceptionOnDuplicateData
+     */
+    public function testParseExceptionOnDuplicate($input, $duplicate_key)
+    {
+        $deprecations = array();
+        set_error_handler(function ($type, $msg) use (&$deprecations) {
```

Comment:
use `ErrorAssert::assertDeprecationsAreTriggered` instead (grep the code to find usages)